src/preconnect.js

```diff
@@ -283,7 +283,8 @@ class PreconnectService {
     // Unfortunately there is no reliable way to feature detect whether
     // preconnect is supported, so we do this only in Safari, which is
     // the most important browser without support for it.
-    if (this.features_.preconnect || !this.platform_.isSafari()) {
+    if (this.features_.preconnect ||
+        !(this.platform_.isSafari() || this.platform_.isIos())) {
```

Comment:
There is a `isWebkit` on `platform` service.

Reply:
It looks like that checks for "WebKit" in the user agent:

https://github.com/ampproject/amphtml/blob/8abab9822dd430a806a96087536c2ae1a8f5072e/src/service/platform-impl.js#L109-L111

But this matches Chrome (both desktop and mobile). For example, my Chrome user agent is `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.79 Safari/537.36`

Perhaps that should be changed to exclude Chrome (but then also force include iOS)? I see only one usage currently of `isWebKit()`:

https://github.com/ampproject/amphtml/blob/acc1927661eec3e7bea03456971c2cb7f83ffe4d/src/service/viewport/viewport-binding-natural.js#L249

I could change `isWebkit()`, then change that usage to `isWebKit() || isChrome()`

I guess there is also a question of what is considered as "WebKit". For example, would you consider Chrome pre-Blink as WebKit? I suggested above having a function named `isActuallySafari` (could use a better name) to make it a bit more clear.

Reply:
are you suggesting `isActuallySafari` == `platform.isSafari() || platform.isIos()` or changing `isSafari` to be that? (we do have some `isSafari` checks that rely on actually safari behaviour like bottomnavr bar). 

`platform.isSafari() || platform.isIos()` seems fine to me, can't think of a better name for it than `isSafariOrIos`.